rules/klayout/drc/testing/run_switch_checking.py

```diff
@@ -69,61 +70,103 @@ def run_switches():
     LOG_FILE_NAME = "run.log"
 
     def run_test_case(test_case, switches, expected_result,expected_log):
-        test_case = test_case.split("--")[-1]
-        run_directory = RUN_DIRECTORY + test_case
-        os.system('mkdir -p {0}'.format(run_directory))
-
-        log_file = run_directory + "/" + test_case + ".log"
-        os.system('python3 ../run_drc.py --path=switch_checking/simple_por.gds.gz {0} >> {1}'.format(switches, log_file))
-
-        check_output_log(log_file, expected_result)
-
-    def check_output_log(log_file, expected_result):
+        try:
+            test_case = test_case.split("--")[-1]
+            run_directory = RUN_DIRECTORY + test_case
+            check_call(f'mkdir -p {run_directory}', shell=True)
+
+            log_file = run_directory + "/" + test_case + ".log"
+            check_call(f'python3 ../run_drc.py --path=switch_checking/simple_por.gds.gz {switches} >> {log_file}', shell=True)
```

Comment:
curious if `run_dir` could be imported as a regular python function and called rather than being run from the shell?

that would allow you to get better structured result, rather than having the perform extra processing on the output logs.